libpyvinyl/Parameters/Collections.py

```diff
@@ -0,0 +1,316 @@
+import json
+from libpyvinyl.AbstractBaseClass import AbstractBaseClass
+from .Parameter import Parameter
+
+
+class Parameters(AbstractBaseClass):
+    """
+    Collection of parameters related to a single calculator
+
+    Parameters are stored in a dict using their name as key
+    """
+    def __init__(self, parameters=None):
+        """
+        Creates a Parameters object, optionally with list of parameter objects
+        """
+        self.parameters = {}
+        if parameters is not None:
+            self.add(parameters)
+
+    def check_type(self, parameter):
+        """
+        Checks given parameter is of type Parameter
+        """
+        if not isinstance(parameter, Parameter):
+            raise RuntimeError(
+                "A non-Parameter object was given to Parameters class.")
+
+    def check_list_type(self, parameter_list):
+        """
+        Checks given list of parameters is a list and contains only parameter objects
+        """
+        if not isinstance(parameter_list, list):
+            raise RuntimeError("Parameters class takes list as input.")
+
+        for par in parameter_list:
+            self.check_type(par)
+
+    def add(self, parameter):
+        """
+        Adds parameters to this parameters object, either single or list
+        """
+        # handle case where list of parameters given
+        if isinstance(parameter, list):
+            self.check_list_type(parameter)
+            for par in parameter:
+                if par.name in self.parameters:
+                    raise RuntimeError(
+                        "Duplicate parameter name in parameters!")
+
+                self.parameters[par.name] = par
+            return
+
+        # handle case where single parameter is given
+        self.check_type(parameter)
+        if parameter in self.parameters:
+            raise RuntimeError("Duplicate parameter name in parameters!")
+
+        self.parameters[parameter.name] = parameter
+
+    def new_parameter(self, *args, **kwargs):
+        """
+        Creates a new parameter with given arguments and adds to this Parameters object
+        """
+        new_parameter = Parameter(*args, **kwargs)
+        self.add(new_parameter)
+
+    def __getitem__(self, key):
+        """
+        Gets parameter with given name from internal dict
+        """
+        return self.parameters[key]
+
+    def __setitem__(self, key, value):
+        """
+        Sets value of parameter with given key to given value
+        """
+        self.parameters[key].set_value(value)
+
+    def __delitem__(self, key):
+        """
+        Deletes parameter with given key
+        """
+        del self.parameters[key]
+
+    def print_indented(self, indents):
```

Comment:
could be made private since only a helper method

Reply:
Hi @CFGrote ,
I've made `print_indented`. Please check it here: https://github.com/PaNOSC-ViNYL/libpyvinyl/pull/17/commits/afd63cc898a4366b896eead855a2dc73ca31eb37

Reply:
Actually, I just found it was called in `ParametersCollection`, so we cannot make it private.
https://github.com/PaNOSC-ViNYL/libpyvinyl/blob/6ee07ae93a6f6638dd44da1c74903bff5e4f1b6a/libpyvinyl/Parameters/Collections.py#L316